configuration.py

```diff
@@ -0,0 +1,153 @@
+            self.logger.warn(
+                "The parameter enable_document_permission does not contain a valid value. Expected values are yes/no, found: {}. Considering the default value as yes".format(enable_document_permissions))
+
+        # validating start_time parameter
+        start_time = self.configurations.get('start_time')
+        if not start_time:
+            self.logger.warn(
+                "The parameter start_time is not present. Considering the default value and bringing all the documents since the begining")
+        elif not self.validate_date("start_time", start_time):
+            self.logger.warn(
+                "The parameter start_time is not in the valid format. Considering the default value and bringing all the documents since the begining")
+
+        # validating end_time parameter
+        end_time = self.configurations.get('end_time')
+        if not end_time:
+            self.logger.warn(
+                "The parameter end_time is not present. Considering the default value and bringing all the documents till current time")
+        elif not self.validate_date("end_time", end_time):
+            self.logger.warn(
+                "The parameter end_time is not in the valid format. Considering the default value and bringing all the documents till current time")
+
+        # validating indexing_interval parameter
+        self.validate_interval("indexing_interval",
+                               self.configurations.get('indexing_interval'))
+        # validating deletion_interval parameter
+        self.validate_interval("deletion_interval",
+                               self.configurations.get('deletion_interval'))
+
+        # validating log_level
+        log_level = self.configurations.get('log_level')
+        if not log_level:
+            self.logger.warn(
+                "The parameter log_level is not present. Considering the default value as INFO")
+        elif log_level.lower() not in ['debug', 'info', 'warn', 'error']:
+            self.logger.warn(
+                "The parameter log_level does not contain a valid value. Expected values are debug/info/warn/error, found:{}. Considering the default value as INFO")
+
+        retry_count = self.configurations.get('retry_count')
+        if not retry_count:
+            self.logger.warn(
+                "The parameter retry_count is not present. Considering the default value as 3")
+        else:
+            try:
+                retry_count = int(retry_count)
+            except Exception:
+                self.logger.warn(
+                    "The parameter retry_count does not contain a valid positive integer value, found: {}. Considering the default value as 3".format(retry_count))
+
+        self.logger.info(
+            'Successdully validated all the configuration parameters')
+        return True
+        # validate object and include/exclude fields
+        # TODO
+
+    def get_all_config(self):
```

Comment:
probably a useless function, callers can use `configuration` directly unless you want to protect it.